### PR TITLE
ci(gevent): pin greenlet for legacy gevent versions

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

[Earlier versions of gevent](https://github.com/gevent/gevent/blob/20.9.0/setup.py#L196) (v1.1, v1.2, v1.3, v1.4, v20.9) only specify a minimum greenlet version. greenlet v2.0 updated its ABI to support py3.11 and is only compatible with [gevent>=22.10.2.](https://github.com/gevent/gevent/pull/1922).

This change updates gevent and opentracer tests to  install `greenlet<2` if `gevent<=20.9` is used.

## Checklist
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
are any changes needed to `riotfile.py`?

**Comment by mabdinur** [comment]
> are any changes needed to `riotfile.py`?

Nope, we only use the latest gevent version in riot.

File: tox.ini [hunk 1]
```diff
@@ -177,6 +179,7 @@ deps =
     gevent211: gevent>=21.1,<21.2
     gevent218: gevent>=21.8,<21.9
     gevent228: gevent>=22.8,<22.10
+    greenlet1: greenlet>=1,<2
```
Comment (mabdinur) [inline tox.ini:182]:
```suggestion
    # Note -  gevent<20.12 does not set a maximum supported version. To test with gevent<20.12 we need to manually install greenlet<2.
    greenlet1: greenlet>=1,<2
```

**Review by Kyle-Verhoog — DISMISSED** [review]

**Review by brettlangdon — APPROVED** [review]

**Comment by Kyle-Verhoog** [comment]
@mabdinur opentracing is still failing on gevent: https://app.circleci.com/pipelines/github/DataDog/dd-trace-py/23030/workflows/a85977a7-c9b0-42d5-88c6-bd4eaee54b4a/jobs/1566740

**Comment by mabdinur** [comment]
@Mergifyio backport 1.6 1.5 1.4 0.61

**Review by Kyle-Verhoog — APPROVED** [review]

**Review by Yun-Kim — APPROVED** [review]

**Comment by jd** [comment]
@mabdinur 
I looked into the profiling failure and there's something very odd. Basically, the `atexit` of uwsgi is interrupted abruptly at a random point which prevents the profiler to flush out the profile. I'm not sure what happens under the hood, if a greenlet is never resumed or something.

**Pull request merged**